### DISCUSSION TITLE
[ Device-SDK-C ] Please help in understanding a few queries on how services built with C work ?

### DISCUSSION THREAD

**Discussion by M0hanrajp** [edgexfoundry/community#305] [Q&A]
### Questions:

* How to run edgex with devices that are built using device-sdk-c ?
  * Per example provided from [Device Service SDK - Getting Started C SDK](https://docs.edgexfoundry.org/3.1/microservices/device/sdk/devicesdk-getting-started/GettingStartedSDK-C/) it seems hyrbrid work environment is used.
  * What do I need to consider while running applications built on C with edgex ?
* How can I eliminate errors such as [Curl failed with code 6(Couldn't resolve host name)](https://github.com/orgs/edgexfoundry/discussions/303) 
  * Is there any configuration change that I need to do ? based on the already found [work-around](https://github.com/orgs/edgexfoundry/discussions/303#discussioncomment-10392388) I don't think it's the actual solution required for running device services that are built using device-sdk-c.
* Why is there no UI update on the nodes in consul UI ? for any device-service.

For example if I run device-random with edgex services:
I can see on localhost:8500, services tab, device-random shows green check.
![image](https://github.com/user-attachments/assets/a7e8bda7-df9e-4cfa-b324-e8bac0ec09ac)
Now If I check `Nodes` tab, then I do not see the device-random service running here.
![image](https://github.com/user-attachments/assets/3f0123a4-aea2-4b1d-81ef-997c0f8a655d)

Why is there no service observed on the nodes tab ? is this based on implementation specific ? can I add this functionality using the current device-sdk-c ?

Please help in understanding the above !

---
Thanks

**Comment by jiekechoo** [comment]
Wrting custom device-service please use the guide: https://docs.edgexfoundry.org/3.1/microservices/device/sdk/devicesdk-getting-started/GettingStartedSDK-C/ 

There are a couple of device services that have been open source here: https://docs.edgexfoundry.org/3.1/microservices/device/DeviceService/

The EdgeX device-service is not a device, it is a collector for IoT device/machine.

**Reply by M0hanrajp** [reply]
Thanks for your response !, could you please look into other questions I have posted above because
I'm aware of the information posted by you, I just need clarifications on specific questions I've rasied.
